platform/ios/src/MGLMapView.h

```diff
@@ -286,6 +286,19 @@ MGL_EXPORT
  */
 - (IBAction)reloadStyle:(nullable id)sender;
 
+/**
+ A boolean value that indicates if whether the map view should automatically
+ adjust its content insets.
+ 
+ When this property is set to `YES` the map automatically updates its
+ `contentInset` property to account for any area not covered by navigation bars,
+ tab bars, toolbars, and other ancestors that obscure the map view.
+ 
+ The default value of this property is `YES`.
+ 
+ */
+@property (assign) BOOL automaticallyAdjustContentInset;
```

Comment:
Let's reconsider using existing contentInsetAdjustmentBehavior instead of adding new property.

```
@property(nonatomic,assign) BOOL automaticallyAdjustsScrollViewInsets API_DEPRECATED("Use UIScrollView's contentInsetAdjustmentBehavior instead", ios(7.0,11.0),tvos(7.0,11.0)); // Defaults to YES
```

Reply:
Yes, please — this would address #12143.

Reply:
@astojilj @friedbunny `automaticallyAdjustsScrollViewInsets` is a property that belongs to a controller which all views are added to. If we use the `UIScrollView`s version then we will consider that property only if the root view of a `mapView` is a `scrollView` which may not be the case.

It is a problem using an Apple SDK property to control the behavior of the map's sdk because we can't add a proper deprecation notice for it.

Reply:
In my comment https://github.com/mapbox/mapbox-gl-native/pull/15584/files#r322363031 I stated why using the API described in https://github.com/mapbox/mapbox-gl-native/pull/15584/files#r322082400 is not feasible.

I am keeping the property, and supporting the previous `automaticallyAdjustsScrollViewInsets`

Reply:
Rename this property to `automaticallyAdjustsContentInset`, so it looks like a property rather than an action method.

Reply:
The new UIScrollView property’s type is [an enumeration](https://developer.apple.com/documentation/uikit/uiscrollview/contentinsetadjustmentbehavior) rather than a Boolean. Do we anticipate adding more possible values in the future?

Reply:
> The new UIScrollView property’s type is an enumeration rather than a Boolean. Do we anticipate adding more possible values in the future?

@1ec5 no, do you have in mind any use case that will require an enumeration instead?